### PR TITLE
Support for description in imgtable

### LINKED ISSUE #1136: Add additional description in component overview documentation
In the documentation are a few pages showing an overview of components: sensors, switches, outputs, etc. 

The numbers of different components aare ever increasing and having only a name and a (small) picture makes it difficult to "browse". Beginners will not always know the exact component code they are looking for, and even seasoned hobbyist will not know every kind of temperature sensor there is. 

I suggest to add a row below picture and name that can hold a simple description, e.g. "Temperature" or "Port expander". Maybe add more information like "Temperature, I2C".

Obviously one should check the component page for more information, but this description would help narrowing down the search at a glance. 

Happy to make a PR for this, but maybe I'll just wait and hear if this would be a welcome addition? I could not find anything in the past issues or requests. 

Esphome is easily my favourite software project and my home is full of different nodes. Here and there a few custom modules and makeshift solutions. I like to read the changelogs and recheck the components overviews in the documentation. Maybe an idea for a new node will pop up. Often I find myself reading the page for yet another temperature sensor, but I have enough of those :)

### PR DESCRIPTION
**Pull request by Grtschnk**
## Description:

Edit: Oh boy, not sure if I got the branch selection right? Not intending to commit the whole backlog, just my last two commits.

Python function ImageTableDirective for imgtable extended to allow further optional arguments. Arguments merged into description string, displayed underneath link/name. As of now, it creates an additional paragraph inside the table cell, might take up too much space. The alternative would be appending the text with a newline (or adjusting CSS). Feel free to adjust or let me know.

**Related issue (if applicable):** fixes https://github.com/esphome/feature-requests/issues/1136

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** n/a<esphome PR number goes here>

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`. Using 'next' branch as it is a new feature (and design adjustment) for the documentation.
  - [ ] **n/a** Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Review by frenck — CHANGES_REQUESTED** [review]
Hi there @Grtschnk 👋 

Fixed the target branch, so it looks better now 😉 

I don't see where this change should so up?
Also, there is a merge conflict, could you take a look at resolving that?

Thanks! 👍 

../Frenck

**Comment by Grtschnk** [comment]
The change is meant to affect the list of sensors (and any other list entry):
https://esphome.io/index.html#sensor-components

Underneath the sensor name an additional optional text, i.e. description, is displayed. For a handful of sensors in index.rst I added the type of sensor. E.g. "Environment" for all those sensors that combine temperature and humidity (and possibly pressure). 

People who want to integrate a specific sensor will know what it is they are using. But some people (including me) like to browse this website to check what great new functionalities esphome has. I already have an environment sensor, so can I quickly ignore these, but a new sensor type altogether would be directly visible.

Also described in https://github.com/esphome/feature-requests/issues/1136 (got automatically closed)

**Comment by frenck** [comment]
The reason I ask: If I look at the build of this branch:

image

I would have expected to see something 😬 

See for yourself: <https://deploy-preview-1016--esphome.netlify.app/#sensor-components>

**Comment by Grtschnk** [comment]
~~My older version works as intended, but when I try to run the latest current branch (in docker) I get an error about a missing "schema.json"file in "schema_doc.py". Seems like this is a recent addition, as I did not have it in my old version.~~

~~Now I tried looking around, but I don't know what exact schema.json file I need: not sure if it is something public,~~ or if it just didn't make it into the repository?

Edit: Okay, I need to generate the json from the main esphome repo (lost track with all those repos). Will get onto it and update.

As far as I can see, the aim of this json schema is to parse information better into the docs. Maybe this interferes with my own addtions.

To give a visual example: That's my local screenshot:
image

**Comment by Grtschnk** [comment]
Of course, it wouldn't work. 
My earlier commit missed lines I had locally. Fixed it.

File: github.py [hunk 1]
```diff
@@ -111,15 +111,19 @@ class ImageTableDirective(Table):
     }
 
     def run(self):
-        cols = self.options.get("columns", 3)
+        cols = self.options.get("columns", 4)
```
Comment (ludeeus) [inline github.py]: IMO 3 is already pushing it for mobile, until there can be a different amount of columns for desktop/mobile I don't think this should change.

Comment (frenck) [reply github.py]:
```suggestion
        cols = self.options.get("columns", 3)
```

**Pull request closed without merging**